version/version.go

```diff
@@ -17,14 +17,124 @@
 // Package version represents the version the project Dragonfly.
 package version
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"text/template"
+)
+
 var (
-	// Version is the version of the project Dragonfly
+	// version is the version of project Dragonfly
+	// populate via ldflags
+	version string
+
+	// revision is the current git commit revision
+	// populate via ldflags
+	revision string
+
+	// branch is the current git branch
+	// populate via ldflags
+	branch string
+
+	// buildDate is the build date of project Dragonfly
 	// populate via ldflags
-	Version string
+	buildDate string
+
+	// goVersion is the running program's golang version
+	goVersion = runtime.Version()
+
+	// goOS is the running program's operating system
+	goOS = runtime.GOOS
+
+	// goArch is the running program's architecture target
+	goArch = runtime.GOARCH
 
 	// DFDaemonVersion is the version of dfdaemon
-	DFDaemonVersion = Version
+	DFDaemonVersion = version
 
 	// DFGetVersion is the version of dfget
-	DFGetVersion = Version
+	DFGetVersion = version
+
+	// SupernodeVersion is the version of supernode
+	SupernodeVersion = version
+
+	// DFVersion the global instance of DragonflyVersion
+	DFVersion *DragonflyVersion
 )
+
+func init() {
+	DFVersion = &DragonflyVersion{
+		version,
+		revision,
+		branch,
+		buildDate,
+		goVersion,
+		goOS,
+		goArch,
+	}
+}
+
+// versionInfoTmpl contains the template used by Info.
+var versionInfoTmpl = `
+{{.program}}, version  {{.version}}
+  git  branch:      {{.branch}}
+  git  commit:      {{.revision}}
+  build  date:      {{.buildDate}}
+  go  version:      {{.goVersion}}
+  go       os:      {{.goOS}}
+  go     arch:      {{.goArch}}
+`
+
+// Print returns version information.
+func Print(program string) string {
+	m := map[string]string{
+		"program":   program,
+		"version":   version,
+		"branch":    branch,
+		"revision":  revision,
+		"buildDate": buildDate,
+		"goVersion": goVersion,
+		"goOS":      goOS,
+		"goArch":    goArch,
+	}
+	t := template.Must(template.New("version").Parse(versionInfoTmpl))
+
+	var buf bytes.Buffer
+	if err := t.ExecuteTemplate(&buf, "version", m); err != nil {
+		panic(err)
+	}
+	return strings.TrimSpace(buf.String())
+}
+
+// DragonflyVersion contains build information about Dragonfly.
```

Comment:
You don't need to redefine `DragonflyVersion ` because it  will be generated by go-swagger which locates at `apis/types/dragonfly_version.go`.